.github/workflows/product_builder.yaml

```diff
@@ -209,8 +216,35 @@ jobs:
         run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
       - name: Run unit tests with coverage
         run: make test-unit-with-coverage
+      - name: Upload unit test results
+        uses: actions/upload-artifact@v3
+        with:
+          name: golang-unit-coverage
+          path: ${{ env.GOLANG_WORKING_DIRECTORY }}/**.cov
+  go_coverage_upload:
+    runs-on: ubuntu-22.04
+    needs:
+      - go_security
+      - go_lint
+      - go_test
```

Comment:
If eg. `go_test` needs `gather_changes` is there any reason why we are adding in here? Or just a double check?

Reply:
No, I think it has no use. I'm removing it.